### PR TITLE
created api routes for generation of all reports and a specific report

### PR DESCRIPTION
**Pull request by HussainKhanSherwani**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes

### CODE REVIEW

File: ai-agents/crowd-fund-analysis/cf_analysis_agent/controller.py [hunk 1]
```diff
@@ -110,7 +116,7 @@ async def upload_to_s3(content, s3_key, content_type="text/plain"):
         Key=f"crowd-fund-analysis/{s3_key}",
         Body=content,
         ContentType=content_type,
-        ACL="public-read",
+        # ACL="public-read",
```
Comment (RobinNagpal) [inline ai-agents/crowd-fund-analysis/cf_analysis_agent/controller.py:119]: Remove this

File: ai-agents/crowd-fund-analysis/cf_analysis_agent/app.py [hunk 2]
```diff
@@ -78,6 +128,86 @@ def commit_info():
...
     return render_template("commit_info.html", commit_hash=commit_hash, commit_message=commit_message)
 
+@app.route('/api/projects/<projectId>/reports/regenerate', methods=['POST'])
+def regenerate_reports(projectId):
+    """
+    Regenerates reports for a given project using values from agent-status.json in S3.
+    """
+    try:
+        # Extract variables from S3
+        variables = extract_variables_from_s3(projectId)
```
Comment (DawoodMehmood) [inline ai-agents/crowd-fund-analysis/cf_analysis_agent/app.py:138]: this logic is being repeated in the other route as well so might as well extract it out to avoid repetition

File: ai-agents/crowd-fund-analysis/cf_analysis_agent/financial_review_agent.py [hunk 3]
```diff
@@ -13,9 +13,16 @@
 
 load_dotenv()
 
+OPENAI_MODEL = os.getenv("OPENAI_MODEL")
```
Comment (DawoodMehmood) [inline ai-agents/crowd-fund-analysis/cf_analysis_agent/financial_review_agent.py:16]: model initialization and llm creation is being repeated in each report so might as well extract it out in a helper or utilities file
